src/front/glsl/parser.rs

```diff
@@ -46,6 +50,15 @@ impl<'source, 'program, 'options> Parser<'source, 'program, 'options> {
         self.lexer.next().ok_or(ErrorKind::EndOfFile)
     }
 
+    fn bump_if(&mut self, value: TokenValue) -> Option<Token> {
+        // Ignore end of file
+        if self.lexer.peek().map_or(false, |t| t.value == value) {
+            self.bump().ok()
```

Comment:
We can do something like
```rust
self.lexer.peek().filter(|t| t.value == value).and_then(|_| self.bump().ok())
```

Reply:
the borrow checker can't see through this :slightly_frowning_face: 
```
   |
55 |         self.lexer
   |         ---------- borrow occurs here
...
58 |             .and_then(|_| self.bump().ok())
   |              -------- ^^^ ---- second borrow occurs due to use of `self` in closure
   |              |        |
   |              |        closure construction occurs here
   |              first borrow later used by call
```

Reply:
oh sweet! https://doc.rust-lang.org/std/iter/struct.Peekable.html#method.next_if